### PR TITLE
No more 6.3 compatibility + Use default files permissions from Elasticsearch package

### PR DESCRIPTION
**Pull request by jmlrt**
- Cleanup legacy code related to Elasticsearch versions < 6.3 => Fix https://github.com/elastic/ansible-elasticsearch/issues/607

- Also use files permissions from official packages instead of changing them => Fix https://github.com/elastic/ansible-elasticsearch/issues/430

### CODE REVIEW

File: tasks/xpack/security/elasticsearch-security-file.yml [hunk 1]
```diff
@@ -1,7 +1,10 @@
 ---
-- name: set fact manage_file_users
-  set_fact: manage_file_users=es_users is defined and es_users.file is defined and es_users.file.keys() | list | length > 0
+- set_fact: manage_file_users=false
 
+- set_fact: manage_file_users=true
+  when: es_users is defined and es_users.file is define and es_users.file.keys() | list | length > 0
```
Comment (Crazybus) [inline tasks/xpack/security/elasticsearch-security-file.yml]:
```suggestion
  when: es_users is defined and es_users.file is defined and es_users.file.keys() | list | length > 0
```

This fixes the currently failing test:

```
14:34:57        TASK [elasticsearch : set_fact] ************************************************
14:34:57        fatal: [localhost]: FAILED! => {"msg": "The conditional check 'es_users is defined and es_users.file is define and es_users.file.keys() | list | length > 0' failed. The error was: template error while templating string: no test named 'define'. String: {% if es_users is defined and es_users.file is define and es_users.file.keys() | list | length > 0 %} True {% else %} False {% endif %}\n\nThe error appears to have been in '/tmp/kitchen/elasticsearch/tasks/xpack/security/elasticsearch-security-file.yml': line 4, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- set_fact: manage_file_users=true\n  ^ here\n"}
14:34:57        
```

Comment (jmlrt) [reply tasks/xpack/security/elasticsearch-security-file.yml]: Fixed in https://github.com/elastic/ansible-elasticsearch/pull/609/commits/08512fc17ea49568a58f788d1ed998133a68fd3e

**Comment by jmlrt** [comment]
@Crazybus regarding the new failing test:

```
         1) Xpack upgrade Tests File "/etc/elasticsearch/users" should be owned by "elasticsearch"
            Failure/Error: it { should be_owned_by 'elasticsearch' }
              expected `File "/etc/elasticsearch/users".owned_by?("elasticsearch")` to return true, got false
              /bin/sh -c stat\ -c\ \%U\ /etc/elasticsearch/users\ \|\ grep\ --\ \\\^elasticsearch\\\$
````

`/etc/elasticsearch/users` file comes from the default RPM with `root` owner.
Should we force `chown` to `elasticsearch` user in this role or let it owned by `root` and remove test on the ownership?

IMO I prefer the 2nd solution. We should keep default RPM settings when we can. WDYT?

**Comment by Crazybus** [comment]
> IMO I prefer the 2nd solution. We should keep default RPM settings when we can. WDYT?

Yup I agree! I think its probably a leftover from the multi instance support.

**Comment by jmlrt** [comment]
@Crazybus This is ready for new review.
I took advantage of it to change files permissions to the ones set by official Elasticsearch packages.

This should also fix https://github.com/elastic/ansible-elasticsearch/issues/430.

**Pull request merged**